config/config.go

```diff
@@ -21,15 +21,51 @@
 package config
 
 import (
+	"time"
+
 	"github.com/uber-go/dosa"
 	"github.com/uber-go/dosa/connectors/yarpc"
 )
 
+const (
+	_defaultConn = "yarpc"
+
+	// default search location and what to exclude
+	_defEntityPath = "./entities/dosa"
+	_defExclude    = "_test.go"
+
+	// default timeouts in milliseconds
+	_defInitializeTimeout        = 10000
+	_defCreateIfNotExistsTimeout = 10000
+	_defReadTimeout              = 10000
+	_defUpsertTimeout            = 10000
+	_defRemoveTimeout            = 10000
+	_defRangeTimeout             = 10000
+	_defSearchTimeout            = 10000
+	_defScanEverythingTimeout    = 10000
+)
+
+// TimeoutConfig holds timeout values for all DOSA operations
+type TimeoutConfig struct {
+	Initialize        time.Duration `yaml:"initialize"`
+	CreateIfNotExists time.Duration `yaml:"createIfNotExists"`
+	Read              time.Duration `yaml:"read"`
+	Upsert            time.Duration `yaml:"upsert"`
+	Remove            time.Duration `yaml:"remove"`
+	Range             time.Duration `yaml:"range"`
+	Search            time.Duration `yaml:"search"`
+	ScanEverything    time.Duration `yaml:"scanEverything"`
+}
+
 // Config represents the settings for the dosa client
 type Config struct {
-	Scope      string       `yaml:"scope"`
-	NamePrefix string       `yaml:"namePrefix"`
-	Yarpc      yarpc.Config `yaml:"yarpc"`
+	Scope       string         `yaml:"scope"`
+	NamePrefix  string         `yaml:"namePrefix"`
+	Connector   string         `yaml:"connector"`
+	EntityPaths []string       `yaml:"entityPaths"`
+	Excludes    []string       `yaml:"excludes"`
+	Yarpc       yarpc.Config   `yaml:"yarpc"`
```

Comment:
we should deprecate this field then. And the Yarpc connector should be able to just configure itself. That's beyond the scope of this PR though. When you merge this, I'll get to work on it.